### PR TITLE
Setting cli prompt width to the user's terminal width

### LINKED ISSUE #5467: Set prompt width to the user's terminal width
Today we hardcode the width of a text prompt:

https://github.com/project-radius/radius/blob/main/pkg/cli/prompt/text/text.go#L38

We should instead have a smart approach where we read the user's terminal window set the prompt width based on that instead.

Today if a user's terminal window is below 20 (40 in the current PR), then the window flashes. Since 40 is still a small width this wouldn't be hit too much

[AB#7647](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/7647)

### PR DESCRIPTION
**Pull request by vishwahiremat**
# Description

- Getting the terminal size when the prompt model is created and setting the width to be terminal width.

## Type of change

- This pull request fixes a bug in Radius and has an approved issue (issue link required).
- This pull request adds or changes features of Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: https://github.com/radius-project/radius/issues/5467

## Contributor checklist
Please verify that the PR meets the following requirements, where applicable:

- [ ] An overview of proposed schema changes is included in a linked GitHub issue.
- [ ] A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
- [ ] If applicable, design document has been reviewed and approved by Radius maintainers/approvers.
- [ ] A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
- [ ] A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
- [ ] A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.

### CODE REVIEW

File: pkg/cli/prompt/text/text.go [hunk 1]
```diff
@@ -74,7 +77,12 @@ func NewTextModel(prompt string, options TextModelOptions) Model {
 	// so it will be blocked. This means you can't type `prod-aws` which is a valid name.
 	ti := textinput.New()
 	ti.Focus()
-	ti.Width = 40
+	width, err := getTerminalSize()
+	if err != nil {
+		return Model{}, err
+	}
+
+	ti.Width = width
```
Comment (rynowak) [inline pkg/cli/prompt/text/text.go:85]: Do you know what the behavior is if we don't set the width? https://github.com/charmbracelet/bubbles/blob/master/textinput/textinput.go#L118

Does it autoscale?

Comment (rynowak) [reply pkg/cli/prompt/text/text.go:85]: I agree that hardcoding is not the right thing.

File: pkg/cli/prompt/text/text.go [hunk 2]
```diff
@@ -84,7 +92,7 @@ func NewTextModel(prompt string, options TextModelOptions) Model {
 		options:   options,
 		prompt:    prompt,
 		textInput: ti,
-	}
+	}, nil
```
Comment (rynowak) [inline pkg/cli/prompt/text/text.go:95]: Can you look into this as an approach? https://github.com/charmbracelet/bubbletea/blob/main/examples/progress-download/tui.go#L44

The framework should automatically send us `WindowSizeMsg`

File: pkg/cli/prompt/text/text.go (hunk 1 above)
Comment (vishwahiremat) [reply pkg/cli/prompt/text/text.go:85]: tested it by not setting the width and it automatically sets the width to teminal width and it autoscales as well. I am going to update the changes to remove the setting of width property.

Comment (rynowak) [reply pkg/cli/prompt/text/text.go:85]: Great 👍 simple solutions are good.

File: pkg/cli/prompt/text/text.go [hunk 3]
```diff
@@ -148,7 +151,7 @@ func (m Model) View() string {
 	view := &strings.Builder{}
 	view.WriteString(m.prompt)
```
Comment (rynowak) [inline pkg/cli/prompt/text/text.go:152]: Do we need to implement wrapping for the prompt? or is that done automatically?

**Review by rynowak — APPROVED** [review]
:shipit: looks great! Thanks for finding and fixing the other cases.

**Pull request merged**